lib/schema.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+module.exports = {
+  projects: {
+    url: '/accounts/{accountId}/projects',
+    requiredFields: ['accountId'],
+    list: {
+      type: 'GET'
+    },
+    create: {
+      type: 'POST',
+      requiredFields: ['name', 'sitePrimary']
+    },
+    read: {
+      url: '/{projectId}',
+      type: 'GET',
+      requiredFields: ['projectId']
+    },    
+    update: {
+      url: '/{projectId}',
+      type: 'PATCH',
+      requiredFields: ['projectId']
+    },    
+    delete: {
+      url: '/{projectId}',
+      type: 'DELETE',
+      requiredFields: ['projectId']
```

Comment:
GET and POST /projects without the trailing /{projectId} does not require the projectId, that's why i had them separated a bit.